pkg/apis/eventing/v1/trigger_validation.go

```diff
@@ -43,22 +48,49 @@ type RabbitTrigger struct {
 	Client versioned.Interface
 }
 
-var (
-	_ apis.Validatable = (*RabbitTrigger)(nil)
-)
+var _ apis.Validatable = (*RabbitTrigger)(nil)
 
 func (t *RabbitTrigger) Validate(ctx context.Context) *apis.FieldError {
 	broker, err := t.Client.EventingV1().Brokers(t.Namespace).Get(ctx, t.Spec.Broker, metav1.GetOptions{})
 	if err != nil {
 		return nil
 	}
+
 	bc, ok := broker.GetAnnotations()[eventingv1.BrokerClassAnnotationKey]
 	if !ok || bc != BrokerClass {
 		// Not my broker
 		return nil
 	}
+
+	// if prefetch is set, validate it
+	// if it isn't then the default value (1) is used
+	prefetch, ok := t.GetAnnotations()[prefetchAnnotation]
+	if ok {
+		prefetchInt, err := strconv.Atoi(prefetch)
+		if err != nil {
+			return &apis.FieldError{
+				Message: "Failed to parse valid int from prefetchAnnotation",
+				Paths:   []string{"metadata", "annotations", prefetchAnnotation},
+				Details: err.Error(),
+			}
+		}
+
+		if prefetchInt < 1 || prefetchInt > 1000 {
+			return apis.ErrOutOfBoundsValue(prefetchInt, 1, 1000, prefetchAnnotation)
+		}
+	}
+
 	if apis.IsInUpdate(ctx) {
 		original := apis.GetBaseline(ctx).(*eventingv1.Trigger)
+		origPrefetch, origOk := original.GetAnnotations()[prefetchAnnotation]
```

Comment:
Why make this immutable?

Reply:
So that it behaves the same as the `Source`. I have captured more detail under **Sanity check**: https://github.com/knative-sandbox/eventing-rabbitmq/pull/522#issuecomment-986812472

Reply:
Hmm, it wasn't immutable before and it works correctly when a user changes the value. I don't think just for the sake of consistency we should make immutable, if anything we should drive the source to be mutable as well.

Reply:
I'm up for this change, @gvmw are you ok with it?

Do you prefer a separate PR or in this is good enough?

Reply:
Yes, if this mutable property behaves correctly - which it seems to be the case for `Adapter` - then we should keep as is and make the `Source` behave the same way, meaning support changes and re-connect correctly.